x-pack/test/apm_api_integration/tests/service_overview/instances_main_statistics.spec.ts

```diff
@@ -76,6 +76,96 @@ export default function ApiTest({ getService }: FtrProviderContext) {
+
+          const successfulTimestamps = range.interval('1m').rate(1);
+          const failedTimestamps = range.interval('1m').rate(1);
+
+          const instances = [...Array(200).keys()].map((index) =>
+            apm
+              .service({ name: serviceName, environment: 'production', agentName: 'nodejs' })
+              .instance(`instance-${index}`)
+          );
+
+          const instanceSpans = (instance: Instance) => {
+            const successfulTraceEvents = successfulTimestamps.generator((timestamp) =>
+              instance
+                .transaction({ transactionName })
+                .timestamp(timestamp)
+                .duration(1000)
+                .success()
+                .children(
+                  instance
+                    .span({
+                      spanName: 'GET apm-*/_search',
+                      spanType: 'db',
+                      spanSubtype: 'elasticsearch',
+                    })
+                    .duration(1000)
+                    .success()
+                    .destination('elasticsearch')
+                    .timestamp(timestamp),
+                  instance
+                    .span({ spanName: 'custom_operation', spanType: 'custom' })
+                    .duration(100)
+                    .success()
+                    .timestamp(timestamp)
+                )
+            );
+
+            const failedTraceEvents = failedTimestamps.generator((timestamp) =>
+              instance
+                .transaction({ transactionName })
+                .timestamp(timestamp)
+                .duration(1000)
+                .failure()
+                .errors(
+                  instance
+                    .error({ message: '[ResponseError] index_not_found_exception' })
+                    .timestamp(timestamp + 50)
+                )
+            );
+
+            const metricsets = range
+              .interval('30s')
+              .rate(1)
+              .generator((timestamp) =>
+                instance
+                  .appMetrics({
+                    'system.memory.actual.free': 800,
+                    'system.memory.total': 1000,
+                    'system.cpu.total.norm.pct': 0.6,
+                    'system.process.cpu.total.norm.pct': 0.7,
+                  })
+                  .timestamp(timestamp)
+              );
+
+            return [successfulTraceEvents, failedTraceEvents, metricsets];
+          };
+
+          return synthtrace.index(instances.flatMap((instance) => instanceSpans(instance)));
+        });
+
+        after(() => {
+          return synthtrace.clean();
+        });
+        describe('feetch instances', () => {
+          let instancesMainStats: ServiceOverviewInstancesMainStatistics['currentPeriod'];
+          before(async () => {
+            instancesMainStats = await getServiceOverviewInstancesMainStatistics({
+              serviceName,
+            });
+          });
+          it('returns top 100 instances', () => {
```

Comment:
Is it worth it to add another test case to validate the sorting feature?

Reply:
I added the remaining tests here https://github.com/elastic/kibana/pull/174164/files#diff-e71c599b58d608588b56018bec35384d06323dc2e7f060b4ce017a923c06fa96R419, it's hard to predict what'll be the exact result of each variable with that amount of instances. In the link above I have control over the items created.